build.sbt

```diff
@@ -423,29 +419,35 @@ lazy val crossVersionSharedSources: Seq[Setting[_]] =
   Seq(Compile, Test).map { sc =>
     (sc / unmanagedSourceDirectories) ++= {
       (sc / unmanagedSourceDirectories).value.map { dir: File =>
-        CrossVersion.partialVersion(scalaBinaryVersion.value) match {
+        CrossVersion.partialVersion(scalaVersion.value) match {
           case Some((major, minor)) =>
             new File(s"${dir.getPath}_$major.$minor")
           case None =>
             sys.error("couldn't parse scalaBinaryVersion ${scalaBinaryVersion.value}")
         }
       }
     }
-  }
+  } ++ Seq(
+    Compile / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("main",
+                                                                         baseDirectory.value,
+                                                                         scalaVersion.value
+    ),
+    Test / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("test", baseDirectory.value, scalaVersion.value)
+  )
```

Comment:
I tried removing this whole block `lazy val crossVersionSharedSources` locally and it seems that there are no issues.